### PR TITLE
wire up 'sign on' buttons for questions

### LINKED ISSUE #38: wire up 'sign on' button for questions
Clicking on the 'sign on' button for a question should:
- record the action in the database
- update signature counts
- give user feedback that they have successfully signed on

Prerequisite for #20

### PR DESCRIPTION
**Pull request by derekeder**
Clicking 'sign on' button triggers javascript AJAX call to create new signature with current_user and selected question. Upon success, javascript toggles button text, disables button, and updates signature count.

We went with the AJAX approach to benefit the user interaction on question list view. This allows users to quickly scan a question, sign on to it, and carry on. Reused this code (https://github.com/datamade/opengovernment-app/blob/master/app/assets/javascripts/sign_on_user.js) for the question show view as well.

Additionally, the signature progress bar now renders with the appropriate width % attribute.

Todo:
- need button styles for 'signed on' button
- spec: @walter what signature functionality would you like to see tested?

Closes #38

### CODE REVIEW

File: app/assets/javascripts/sign_on_user.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+$('.sign-on-user').click(function(){
```
Comment (walter) [inline app/assets/javascripts/sign_on_user.js]: So far we have been using coffeescript on this project, unless you have a compelling reason to use JS for this, please redo as coffeescript to be consistent with the conventions of the project.

File: app/models/user.rb [hunk 2]
```diff
@@ -108,6 +108,11 @@ def questions_signed
     Question.find(signatures.map(&:question_id))
   end
 
+  # @return [Boolean] a given question has been signed by the user
+  def question_signed(question_id)
```
Comment (walter) [inline app/models/user.rb]: Ruby style guide:

> The names of predicate methods (methods that return a boolean value) should end in a question mark. (i.e. Array#empty?).

**Comment by walter** [comment]
I'm not seeing any corresponding acceptance tests added for this feature to spec/requests/questions_spec.rb. I also see a failure now happens on the index action.

Have a crack at adding the necessary acceptance tests and also fixing the failing test. Generally looks good though.

Let me know if you need a hand with the tests.

**Comment by walter** [comment]
Ah, I see you actually asked what I would like to have tested.

Basically it boils down to at least a test of the most common path through the feature being added. If there is more than one common path (e.g. signed in versus signed out) than covering them is good.

I.e. if you are adding a new feature, I would like an acceptance test (aka an integration test or under "spec/requests/") that confirms that it is working as described above.

In this case, I would say that the user can click the button, get the signature/sign up form, and submit and have their signature counted in the signature count.

I would have a similar test for logged in users being able to add their signature to the question.

Make sense?

**Comment by derekeder** [comment]
@walter spec outline makes sense. got the current tests to pass

Will work on some basic sign on tests.

**Comment by derekeder** [comment]
@walter all right I think this is ready to merge in

There was a form built out for unregistered users to sign on to a question, but the signature creation was not implemented. It required overriding the devise user create method, which I went ahead and did in 2d4adc3. 

Tests are implemented and passing for both signed in and unregistered users.

**Pull request merged**